### PR TITLE
Refactor integrations based on recent work and reviews on Jira integration

### PR DESCRIPTION
**Pull request by daabr**
Sorry for lumping these together, but there was no way to split them cleanly.

* Add `ConnectionCapabilities` to integrations that require initialization by the user
* Simplification of content-type check
* `redirectToErrorPage` function, instead of many `fmt.Sprintf` calls
* Replace all instances of `uiPath` consts with `desc.ConnectionURL().Path`
* Add link to Python client API
* GitHub: simplify `connStatus` function, multiple installs for a single connection are no longer possible, since we're not using tokens anymore

New additions:

* GitHub: fix ENG-991
* HTTP mux paths: add Go 1.22's HTTP method specification
* GitHub: replace per-file URL paths with Go 1.22's pattern wildcard
* Move all URL path consts to `server.go` - where their primary usage is
* Reduce multiple calls to `r.Context()` with a helper var `ctx`
* Replace references to `valuesv1.Value` with `sdktypes.ValuePB`
* Construct SDK event objects before, not during, dispatches
* Better code reused in Slack event handlers

### CODE REVIEW

File: integrations/github/client.go [hunk 1]
```diff
@@ -68,17 +69,9 @@ func connStatus(i *integration) sdkintegrations.OptFn {
 
 		if vs.Has(vars.PAT) {
 			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "using PAT"), nil
+		} else {
+			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "using OAuth 2.0"), nil
```
Comment (itayd) [inline integrations/github/client.go:73]: drop 2.0

File: integrations/github/webhooks/webhook.go [hunk 2]
```diff
@@ -203,38 +185,40 @@ func extractInstallationID(event any) (inst string, err error) {
...
+			return
+		}
 
-		e := kittehs.Must1(sdktypes.EventFromProto(event))
 		eventID, err := d.Dispatch(ctx, e, nil)
 		if err != nil {
-			l.Error("Dispatch failed", zap.Error(err))
+			l.Error("Event dispatch failed",
+				zap.String("eventID", eventID.String()),
+				zap.String("connectionID", cid.String()),
```
Comment (itayd) [inline integrations/github/webhooks/webhook.go]: use With

File: integrations/jira/event.go [hunk 3]
```diff
@@ -59,26 +59,30 @@ func (h handler) handleEvent(w http.ResponseWriter, r *http.Request) {
...
 	if !strings.HasPrefix(contentType, contentTypeJSON) {
+		// This is probably an attack, so no user-friendliness.
 		http.Error(w, "Bad Request", http.StatusBadRequest)
 		return
 	}
 
 	// Parse some of the metadata in the Jira event's JSON content.
 	body, err := io.ReadAll(r.Body)
 	if err != nil {
+		// This is probably an attack, so no user-friendliness.
```
Comment (itayd) [inline integrations/jira/event.go]: not an attack

**Review by itayd — APPROVED** [review]

File: integrations/github/webhooks/webhook.go (hunk 2 above)
Comment (daabr) [reply integrations/github/webhooks/webhook.go]: Fixed across the PR

File: integrations/jira/event.go (hunk 3 above)
Comment (daabr) [reply integrations/jira/event.go]: Fixed across the PR

**Pull request merged**